MoreLinq.Test/OrderedMergeTest.cs

```diff
@@ -0,0 +1,398 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    class OrderedMergeTest
+    {
+        [Test]
+        public void ShouldBeLazy()
+        {
+            var first = new BreakingSequence<object>();
+            var second = new BreakingSequence<object>();
+            Assert.DoesNotThrow(() => MoreEnumerable.OrderedMerge(first, second, id => id, id => id, id => id, id => id, (f, _) => f, null));
```

Comment:
Any specific reason not to use extension method invocation style here and everywhere else? If not, then please rewrite all invocations as `first.OrderedMerge(second, ...)`.